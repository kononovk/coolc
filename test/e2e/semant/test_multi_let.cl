class Main {
    main(): Object {
        let a: Int <- 1,
            b: Int <- 2,
            c: Int <- 3
        in
            a + b + c
    };
};
