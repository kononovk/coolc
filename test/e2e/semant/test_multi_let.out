#1
_program
  #1
  _class
    Main
    Object
    "test/e2e/semant/test_multi_let.cl"
    (
    #2
    _method
      main
      Object
      #3
      _let
        a
        Int
        #3
        _int
          1
        : Int
        #4
        _let
          b
          Int
          #4
          _int
            2
          : Int
          #5
          _let
            c
            Int
            #5
            _int
              3
            : Int
            #7
            _plus
              #7
              _plus
                #7
                _object
                  a
                : Int
                #7
                _object
                  b
                : Int
              : Int
              #7
              _object
                c
              : Int
            : Int
          : Int
        : Int
      : Int
    )
